### PR TITLE
Modify DataSort to use labels from properties if available

### LINKED ISSUE #7282: DataSort - If Data `properties` defines labels these should be used
### Expected Behavior
DataSort should use the labels defined in Data `properties`.

### Actual Behavior
The labels are ignored and the default name is used.

### URL, screen shot, or Codepen exhibiting the issue
https://codesandbox.io/s/grommet-v2-template-forked-5fr4lt?file=/index.js

### PR DESCRIPTION
**Pull request by MikeKingdom**
#### What does this PR do?
DataSort's "sort by" Select lists the fields to sort by labeled by their key in the data, even if the parent Data has `properties` specified. This change modifies the "sort by" Select to label the field to sort by with the label specified in `properties` if available. This is also how the DataFilters and DataColumns already do it.

Original:
image

New:
image

#### Where should the reviewer start?
DataSort.js

#### How should this be manually tested?
Storybook
The DataSort/Simple story has been modified to specify properties.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
Fixes #7282 

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
Yes

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

**Review by jcfilben — COMMENTED** [review]
Code looks good, could we add a unit test?

**Review by jcfilben — APPROVED** [review]
Looks good!

File: src/js/components/DataSort/__tests__/DataSort-test.tsx [hunk 1]
```diff
@@ -84,4 +84,32 @@ describe('DataSort', () => {
...
+    const items = [
+      { name: 'a', loc: 'Home' },
+      { name: 'b', loc: 'School' },
+    ];
+    const properties = {
+      name: { label: 'My Name' },
+      loc: { label: 'Location' },
+    };
+
+    /* const { container } = */ render(
```
Comment (taysea) [inline src/js/components/DataSort/__tests__/DataSort-test.tsx:98]:
```suggestion
    render(
```
remove comment?

**Review by taysea — APPROVED** [review]
Tiny comment on test, otherwise looks good. Let's make sure to capture this appropriately in the release notes since it could result in snapshot updates for consuming teams potentially.

File: src/js/components/DataSort/__tests__/DataSort-test.tsx (hunk 1 above)
Comment (MikeKingdom) [reply src/js/components/DataSort/__tests__/DataSort-test.tsx]: Done

**Pull request merged**